src/ScottPlot/Plottable/RadialGaugePlot.cs

```diff
@@ -0,0 +1,709 @@
+        /// <param name="cy">The y-coordinate of the circle centre</param>
+        /// <param name="text">String to be drawn</param>
+        /// <param name="posPct">Position of text as a percentage of the angle swept</param>
+        /// <seealso href="http://csharphelper.com/blog/2018/02/draw-text-on-a-circle-in-c/"/>
+        protected virtual void DrawTextOnCircle(Graphics gfx, System.Drawing.Font font,
+            Brush brush, RectangleF clientRectangle, float radius, float angleInit, float angleSwept, float cx, float cy,
+            string text, float posPct)
+        {
+            // Modify anglePos to be in the range [0, 360]
+            if (angleInit >= 0)
+                angleInit -= 360f * (int)(angleInit / 360);
+            else
+                angleInit += 360f;
+
+            // Use a StringFormat to draw the middle top of each character at (0, 0).
+            using StringFormat string_format = new StringFormat();
+            string_format.Alignment = StringAlignment.Center;
+            string_format.LineAlignment = StringAlignment.Center;
+
+            // Used to scale from radians to degrees.
+            double RadToDeg = 180.0 / Math.PI;
+            double width_to_angle = 1 / radius;
+
+            // Measure the characters. Use LINQ to add up the character widths.
+            List<RectangleF> rects = MeasureCharacters(gfx, font, clientRectangle, text);
+            var width_query = from RectangleF rect in rects select rect.Width;
+            double text_width = width_query.Sum() / radius;
+
+            // Angular data
+            bool isPositive = angleSwept >= 0;
+            double angle = ReduceAngle(angleInit + angleSwept * (posPct / 100));
+            angle += (1 - 2 * (posPct / 100)) * (isPositive ? 1 : -1) * RadToDeg * text_width / 2; // Set the position to the middle of the text
+
+            bool isBelow = angle < 180 && angle > 0;
+            double theta = angle * Math.PI / 180;
+            theta += (isBelow ? 1 : -1) * text_width / 2;   // Set the position to the beginning of the text
+
+            // Draw the characters.
+            for (int i = 0; i < text.Length; i++)
+            {
+                // Increment theta half the angular width of the current character
+                theta -= (isBelow ? 1 : -1) * rects[i].Width / 2 * width_to_angle;
+
+                // Calculate the position of the upper-left corner
+                double x = cx + radius * Math.Cos(theta);
+                double y = cy + radius * Math.Sin(theta);
+
+                // Transform to position the character.
+                if (isBelow)
+                    gfx.RotateTransform((float)(RadToDeg * (theta - Math.PI / 2)));
+                else
+                    gfx.RotateTransform((float)(RadToDeg * (theta + Math.PI / 2)));
+
+                gfx.TranslateTransform((float)x, (float)y, System.Drawing.Drawing2D.MatrixOrder.Append);
```

Comment:
Can you pass in `x` and `y` instead of `(0, 0)`, and then remove the translation? This would avoid issues around the noncommutativity of matrix multiplication.

Reply:
It makes sense.
However, when I tested your suggestion the gauge labels didn't render.

Reply:
> Can you pass in x and y instead of (0, 0), and then remove the translation? This would avoid issues around the noncommutativity of matrix multiplication.

No because the string still requires the rotation. I refactored this area to be a little more obvious

```cs
double halfPi = Math.PI / 2;
double rotationRadians = isBelow ? theta - halfPi : theta + halfPi;
double rotationDegrees = rotationRadians * RadToDeg;

gfx.RotateTransform((float)rotationDegrees);
gfx.TranslateTransform(x, y, System.Drawing.Drawing2D.MatrixOrder.Append);
gfx.DrawString(text[i].ToString(), font, brush, 0, 0, string_format);
gfx.ResetTransform();
```